### ISSUE TITLE
VOC Classification Code is not working for my model that has the alexnet structure

### ISSUE THREAD

**Issue by Maryames** [philkr/voc-classification#6]
Hi,

I have a pretrained model in torch and I have transfered the weights and biases to the caffe, and I checked if the model is transfered correctly. The problem is that when I want to evaluate my model for pascal dataset with philkr  VOC classification code (https://github.com/philkr/voc-classification), it gives me 7% and it seems that it cannot finetune. I want to use the code with clipping from drop6 instead of drop7 which is the default in the code and I changed the code accordingly, but I still get the same result.I was wondering how I can use the code correctly?I really appreciate any help

Thank you so much,

**Comment by uahsan3** [comment]
Hello, 

I am facing the same problem. I am trying to finetune my model on Pascal VOC 2007 dataset and the loss is not going down beyond random. It's not clear what is going wrong. 

I have tried with default hyperparameter settings as well as my own. 

Please advise. 

Thanks

**Comment by jianghaojun** [comment]
+1

**Comment by jianghaojun** [comment]
I encountered the same problem as @Maryames . And I solved it with the help of @uahsan3 .
If anyone else encounter this problem, please look at the trainval.prototxt generate by the code (in train_cls.py)and check the data layer. The code will create a right data layer and if your alexnet.prototxt also include a data layer, then the network will have two data layer and the image data will not go through the network. Besides, I attach two screenshots of trainval.prototxt below, the first is wrong and the second is right.
![bug1](https://user-images.githubusercontent.com/30039000/50533936-6c6f0000-0b6f-11e9-86cd-2a305a5c80c1.png)

![bug2](https://user-images.githubusercontent.com/30039000/50533937-6da02d00-0b6f-11e9-98c5-a26b4d40742c.png)
